src/vs/editor/common/viewModel/splitLinesCollection.ts

```diff
@@ -1216,8 +1236,11 @@ export class SplitLine implements ISplitLine {
 		if (!this._isVisible) {
 			throw new Error('Not supported');
 		}
-		let startOffset = this.getInputStartOffsetOfOutputLineIndex(outputLineIndex);
-		let endOffset = this.getInputEndOffsetOfOutputLineIndex(model, modelLineNumber, outputLineIndex);
+
+		// These offsets refer to model text with injected text.
+		const startOffset = outputLineIndex > 0 ? this._lineBreakData.breakOffsets[outputLineIndex - 1] : 0;
+		const endOffset = outputLineIndex < this._lineBreakData.breakOffsets.length ? this._lineBreakData.breakOffsets[outputLineIndex] : model.getLineLength(modelLineNumber);
```

Comment:
I think `this._lineBreakData.breakOffsets[this._lineBreakData.breakOffsets.length - 1]` always contains the (text + injected text) length. So I don't think there is a need to go to the `textModel` if `outputLineIndex` is invalid, we could either throw or clamp it to `this._lineBreakData.breakOffsets.length - 1`. Probably clamping would be the safest change.